backend/api/endpoints/users.py

```
@@ -0,0 +1,72 @@
+        username=user.username,
+        email=user.email,
+        password=user.password,
+        role=user.role
+    )
+    db.add(new_user)
+    db.commit()
+    db.refresh(new_user)
+    return new_user
+
+@router.put("/{user_id}", response_model=User)
+async def update_user(user_id: str, user_update: UserUpdate, db: Session = Depends(get_db)):
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)